components/PlantForm/PlantForm.js

```diff
@@ -44,15 +50,38 @@ export default function PlantForm({
     }
 
     if (isEditMode) {
-      onEditPlant({ ...initialData, ...data });
+      onEditPlant({ imageUrl: imageUrl, ...initialData, ...data });
       onCancel();
     } else {
-      onCreatePlant({ ...initialData, ...data });
+      onCreatePlant({ imageUrl: imageUrl, ...initialData, ...data });
     }
 
     event.target.reset();
   }
 
+  async function handleCreateUpload(event) {
+    setIsImageLoading(true);
+
+    event.preventDefault();
+
+    const formData = new FormData();
+    const image = event.target.files[0];
+
+    formData.append("image", image);
+    const response = await fetch("/api/upload", {
```

Comment:
You could consider to implement an error handler if something goes wrong.

Maybe with a try and catch function.